### PR TITLE
Added Rail Fence Cipher Algorithm Issue#2668

### LINKED ISSUE #2668: Rail Fence Cipher in C language[GSSOC'21]
## 💥 Proposal

(A clear and concise description of what the proposal is.)
Rail fence cipher implementation in C programing language. This is a form of transposition cipher which encrypts the plain text into ciphertext by rearranging the characters. I would like to implement this algorithm in C language.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

(Write your answer here.)
Yes, I have read all the contributing guidelines, please assign this issue to me under GSSOC'21.
Thank You.

### PR DESCRIPTION
**Pull request by nagasrihith608**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes

### Description
This algorithm allows us to cipher the plain text using rail fence cipher algorithm
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes #2668
(Write your answer here.)

### CODE REVIEW

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code

**Comment by nagasrihith608** [comment]
Hello, @ankitaggarwal23 I had updated the changes requested, please have a look at it now. Thank you!

**Comment by nagasrihith608** [comment]
@HarshCasper please merge this. Thank you.

File: C/cryptography/railFenceCipher.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/* The rail fence cipher (also called a zigzag cipher) is a form of transposition cipher. It derives its name from the way in which it is encoded. Given a plain-text message and a numeric key, it cipher the given text using Rail Fence algorithm(here key == 2).
```
Comment (HarshCasper) [inline C/cryptography/railFenceCipher.c]: Break this line down into multiple separate lines. That makes it easy for us to read and understand.

File: C/cryptography/railFenceCipher.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/* The rail fence cipher (also called a zigzag cipher) is a form of transposition cipher. It derives its name from the way in which it is encoded. Given a plain-text message and a numeric key, it cipher the given text using Rail Fence algorithm(here key == 2). 
+*/
+#include<stdio.h>
+#include<string.h>
+  
+int main( void ) {
```
Comment (HarshCasper) [inline C/cryptography/railFenceCipher.c]: Don't keep the primary logic of your code in the main() function. Use main() only for the driver code

File: C/cryptography/railFenceCipher.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/*
+ Sample I/O : 
+   
+ Enter your plain text here :Hello World
+ Cipher :HloWrdel ol
+
+ Time complexity  : O(n)  
+ Space complexity : O(n)
+ */
+
```
Comment (HarshCasper) [inline C/cryptography/railFenceCipher.c]: Remove any extra lines

**Pull request closed without merging**